recipes/tls-create-secure-pair-to-tls-socket/src/workflow.ts

```diff
@@ -0,0 +1,236 @@
+	if (edits.length === 0) return null;
+	return rootNode.commitEdits(edits);
+}
+
+function findCreateSecurePairCalls(rootNode: SgNode<JS>): CallSite[] {
+	const pats = [
+		'$X.createSecurePair($A, $B, $C, $D)',
+		'$X.createSecurePair($A, $B, $C)',
+		'$X.createSecurePair($A, $B)',
+		'$X.createSecurePair($A)',
+		'$X.createSecurePair()',
+		'createSecurePair($A, $B, $C, $D)',
+		'createSecurePair($A, $B, $C)',
+		'createSecurePair($A, $B)',
+		'createSecurePair($A)',
+		'createSecurePair()',
+	];
+	const out: CallSite[] = [];
+	for (const p of pats) {
+		const nodes = rootNode.findAll({ rule: { kind: 'call_expression', pattern: p } });
+		for (const n of nodes) {
+			const x = (n as SgNode<JS>).getMatch('X')?.text();
+			const binding = x ? `${x}.createSecurePair` : 'createSecurePair';
+			out.push({ call: n as SgNode<JS>, binding });
+		}
+	}
+	return out;
+}
+
+function buildOptions(a?: string | null, b?: string | null, c?: string | null, d?: string | null): string {
+	const kv: string[] = [];
+	if (a) kv.push(`secureContext: ${a}`);
+	if (b) kv.push(`isServer: ${b}`);
+	if (c) kv.push(`requestCert: ${c}`);
+	if (d) kv.push(`rejectUnauthorized: ${d}`);
+	return `{ ${kv.join(', ')} }`;
+}
+
+function getText(node: SgNode<JS> | undefined): string | null {
+	const t = node?.text()?.trim();
+	return t || null;
+}
+
+function unique<T>(arr: T[]): T[] {
+	return Array.from(new Set(arr));
+}
+
+function renamePairAssignedVariables(rootNode: SgNode<JS>, _bindings: string[]): Edit[] {
+	const edits: Edit[] = [];
+	const decls = rootNode.findAll({
+		rule: {
+			kind: 'variable_declarator',
+			has: {
+				field: 'value',
+				kind: 'call_expression',
+				any: [
+					{ pattern: '$X.createSecurePair($$$ARGS)' },
+					{ pattern: 'createSecurePair($$$ARGS)' },
+				],
+			},
+		},
+	});
+	for (const decl of decls) {
+		const name = decl.field('name');
+		if (name && name.kind() === 'identifier' && name.text() === 'pair') {
+			edits.push(name.replace('socket'));
+		}
+	}
+	return edits;
+}
+
+function rewriteTlsImports(rootNode: SgNode<JS>): Edit[] {
+	const edits: Edit[] = [];
+	const esmNamed = rootNode.findAll({
+		rule: {
+			kind: 'import_statement',
+			all: [
+				{ has: { kind: 'import_clause', has: { kind: 'named_imports' } } },
+				{
+					has: {
+						field: 'source',
+						any: [{ pattern: '\'tls\'' }, { pattern: '"tls"' }, { pattern: '\'node:tls\'' }, { pattern: '"node:tls"' }],
+					},
+				},
+			],
+			not: { has: { kind: 'namespace_import' } },
+		},
+	});
+	for (const decl of esmNamed) {
+		const srcText = decl.field('source')?.text()?.replace(/['"]/g, '') || '';
+		if (srcText !== 'tls' && srcText !== 'node:tls') continue;
+		const named = decl.find({ rule: { kind: 'named_imports' } });
+		if (!named) continue;
+		const specs = named.findAll({ rule: { kind: 'import_specifier' } });
+		const hasCSP = specs.some(s => s.field('name')?.text() === 'createSecurePair');
+		if (!hasCSP) continue;
```

Comment:
I think this just creates an extra set of iteration, so it can be incorporated into the for…of below.